addons/website_slides_survey/controllers/main.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request
+from odoo.osv import expression
+from odoo.addons.website_profile.controllers.main import WebsiteProfile
+
+
+class WebsiteSlidesSurvey(WebsiteProfile):
+    def _prepare_user_profile_values(self, user, **kwargs):
+        """Loads all data required to display the certification attempts of the given user"""
+        values = super(WebsiteSlidesSurvey, self)._prepare_user_profile_values(user, **kwargs)
+        values['show_certification_tab'] = ('user' in values) and (
+            values['user'].id == request.env.user.id or \
+            request.env.user.has_group('survey.group_survey_manager')
+        )
+
+        if not values['show_certification_tab']:
+            return values
+
+        domain = [('survey_id.certification', '=', True)]
+        if values['user'].mapped('email'):
```

Comment:
Why using "mapped"? Can "user" be a recordset of multiple users?

Reply:
With that logic, you will end-up having "mapped" calls literally everywhere in Odoo.

If it's assumed that your recordset contains 0 or 1 user, then you don't use mapped, it's confusing and less convenient (as it will return an array and not the value of the attribute).